graphs/possible_bipartition.py

```diff
@@ -8,5 +25,15 @@ def possible_bipartition(dislikes):
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    """
+    - start with the "first" node
+    - add the node and it's children to a 
+    """
+    
+    for key in dislikes.keys():
```

Comment:
Loops through every dog in the graph. This accounts for the case that the graph contains multiple disjoint cycles. Look at the last test in the test suite for an example where this loop is necessary.

We could optimize this by maintaining a visited list or dictionary that's passed to the helper function and only calling the helper if the node wasn't visited as part of the initial pass.